doc/reference/event_handling.md

```diff
@@ -78,6 +78,15 @@ The `t-on` directive allows to prebind its arguments. For example,
 Here, `expr` is a valid Owl expression, so it could be `true` or some variable
 from the rendering context.
 
+### Type Hinting
+You could also indicate your payload type to enforce your typings. You will see typing errors while developing and resolve it even before it's deployed.
```

Comment:
this section only applies for people working on typescript. This should be explained clearly, otherwise some people will complain that the code from this example does not work in JS.

Maybe add a short sentence to explain, at the beginning of this paragraph: 
```
Note that if you work with Typescript, the `trigger` method is generic on the type of the payload. You can then describe the type of the event, so you will see typing errors...
```

Reply:
Done ;-) What do you think ?